pypesto/storage/read_from_hdf5.py

```diff
@@ -0,0 +1,89 @@
+import h5py
+from ..result import Result
+from ..optimize.result import OptimizerResult
+from ..problem import Problem
+from ..objective import Objective
+
+
+def read_hdf5_optimization(f: h5py.File,
+                           start: h5py.Group) -> 'OptimizerResult':
+    """
+    Read HDF5 results per start.
+
+    Parameters
+    ----------
+    f:
+        The HDF5 result file
+    start:
+        Specifies the start that is read from the HDF5 file
+    """
+
+    result = OptimizerResult()
+
+    for optimization_key in result.keys():
+        if optimization_key in f[f'/optimization/results/{start}']:
+            result[optimization_key] = \
+                f[f'/optimization/results/{start}/{optimization_key}'][:]
+        elif optimization_key in \
+                f[f'/optimization/results/{start}'].attrs:
+            result[optimization_key] = \
+                f[f'/optimization/results/{start}'].attrs[optimization_key]
+    return result
+
+
+class ProblemHDF5Reader:
+    """
+    Reader of the HDF5 problem files written
+    by class ProblemHDF5Writer.
+
+    Attributes
+    ---------
+    storage_filename:
+        HDF5 result file name
+    """
+    def __init__(self, storage_filename: str):
+        self.storage_filename = storage_filename
+        self.problem = Problem(Objective(), [], [])
+
+    def read(self) -> 'Problem':
+        """
+        Read HDF5 problem file and return pyPESTO problem object.
+        """
+        with h5py.File(self.storage_filename, 'r') as f:
+            for problem_key in f['/problem']:
+                setattr(self.problem, problem_key,
+                        f[f'/problem/{problem_key}'][:])
+            for problem_attr in f['/problem'].attrs:
+                setattr(self.problem, problem_attr,
+                        f['/problem'].attrs[problem_attr])
+        return self.problem
+
+
+class OptimizationResultHDF5Reader:
+    """
+    Reader of the HDF5 result files written
+    by class OptimizationResultHDF5Writer.
+
+    Attributes
+    ---------
```

Comment:
parameters, + the "----" must be at least as long as the name before (I know it's silly, but otherwise parsing doesn't work ...)